### PR TITLE
upgrade fancybox

### LINKED ISSUE #1638: Upgrade to Fancybox 4
We're using Fancybox 3. Upgrade to 4.

- [x] Replace files [here](https://github.com/gravityview/GravityView/tree/main/assets/lib/fancybox/dist)
- [x] Modify the [lightbox provider script](https://github.com/gravityview/GravityView/blob/develop/includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php) with corrected attributes, if any need changing

### PR DESCRIPTION
**Pull request by omarkasem**
resolves #1638

### CODE REVIEW

File: includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php [hunk 1]
```diff
@@ -43,9 +43,7 @@ public function print_scripts( $gravityview ) {
 			}
 		</style>
 		<script>
-			if ( window.jQuery ) {
-				jQuery( '.gravityview-fancybox' ).fancybox(<?php echo $settings; ?>);
-			}
+			Fancybox.bind(".gravityview-fancybox", <?php echo $settings; ?>);
```
Comment (zackkatz) [inline includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php:46]: @omarkasem There's always (_always_) some situation where a script doesn't load. Let's wrap in an `if ( window.Fancybox )` condition.

File: includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php [hunk 2]
```diff
@@ -61,6 +59,7 @@ public function print_scripts( $gravityview ) {
 	protected function default_settings() {
 
 		$defaults = array(
+				// 'groupAll'=>true,
```
Comment (zackkatz) [inline includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php:62]: Why was this removed?

File: includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php [hunk 3]
```diff
@@ -94,15 +93,15 @@ protected function default_settings() {
 	 */
 	public function enqueue_scripts() {
 		$min = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
-		wp_register_script( self::$script_slug, plugins_url( 'assets/lib/fancybox/dist/jquery.fancybox' . $min . '.js', GRAVITYVIEW_FILE ), array( 'jquery' ), GV_PLUGIN_VERSION );
+		wp_register_script( self::$script_slug, plugins_url( 'assets/lib/fancybox/dist/fancybox.umd.js', GRAVITYVIEW_FILE ), array( 'jquery' ), GV_PLUGIN_VERSION );
```
Comment (zackkatz) [inline includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php:96]: See my comment on Upwork—in the future, it'd just be good to know more about this type of change.

**Review by zackkatz — CHANGES_REQUESTED** [review]

File: includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php (hunk 2 above)
Comment (omarkasem) [reply includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php]: It was added by mistake while testing for duplicated image isssue

**Comment by zackkatz** [comment]
@omarkasem Sorry for not clarifying; I meant those comments would be great in the commit messages, not as inline comments.

To explain the changes to future developers.

**Comment by mrcasual** [comment]
@omarkasem I could not open PDF & text files. They work on `develop`.

**Comment by zackkatz** [comment]
@omarkasem File types that open in an iframe have `gv-iframe=true` added to the URL, which is then processed here:
https://github.com/gravityview/GravityView/blob/af844d988d6088209bf6f62fed2e4a9f84ce8220/includes/extensions/lightbox/fancybox/class-gravityview-lightbox-provider-fancybox.php#L161-L171

**Comment by zackkatz** [comment]
Closed, since already merged with `develop`. Going to manually cherry-pick b7823a75b8b38ad5e347f04ddccba9d7c27e474f

**Pull request closed without merging**